<h3>Download and install the TensorFlow 2.0 RC package. Import TensorFlow into your program:</h3

In [2]:
from __future__ import absolute_import, division, print_function, unicode_literals

import tensorflow as tf

/home/hackonium/.local/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/hackonium/.local/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/hackonium/.local/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/hackonium/.local/lib/python3.7/site-packages/tensorflow/python/framework/dty

<h3>Load and prepare the MNIST dataset. Convert the samples from integers to floating-point numbers:</h3>

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 15s 1us/step


<h3>Build the tf.keras.Sequential model by stacking layers. Choose an optimizer and loss function for training:</h3>

In [4]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate the model:

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

W0826 23:07:24.413072 140601976829760 deprecation.py:323] From /home/hackonium/.local/lib/python3.7/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 16s 272us/sample - loss: 0.2965 - accuracy: 0.9134
Epoch 2/5
60000/60000 [==============================] - 12s 205us/sample - loss: 0.1421 - accuracy: 0.9578
Epoch 3/5
60000/60000 [==============================] - 13s 223us/sample - loss: 0.1057 - accuracy: 0.9678
Epoch 4/5
60000/60000 [==============================] - 11s 179us/sample - loss: 0.0882 - accuracy: 0.9731
Epoch 5/5
10000/10000 [==============================] - 1s 113us/sample - loss: 0.0791 - accuracy: 0.9759


[0.07905022123379167, 0.9759]

The image classifier is now trained to ~98% accuracy on this dataset.